### PR TITLE
Blog integration specs

### PR DESCRIPTION
**Pull request by megha-n-bodke**
-  The N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe - view: blog all user posts
- Integration specs
- Use Capybara to write integration tests for each view in project.

- User index page:
   - I can see the username of all other users.
   - I can see the profile picture for each user.
   -  I can see the number of posts each user has written.
   -  When I click on a user, I am redirected to that user's show page.
      
- user show page:
   - I can see the user's profile picture.
   - I can see the user's username.
   -  I can see the number of posts the user has written.
   -  I can see the user's bio.
   - I can see the user's first 3 posts.
   -  I can see a button that lets me view all of a user's posts.
   - When I click a user's post, it redirects me to that post's show page.
   -  When I click to see all posts, it redirects me to the user's post's index page.
  
- User post index page:
      - I can see the user's profile picture.
      - I can see the user's username.
      - I can see the number of posts the user has written.
     -  I can see a post's title.
      - I can see some of the post's body.
      -  I can see the first comments on a post.
      - I can see how many comments a post has.
      - I can see how many likes a post has.
     -  I can see a section for pagination if there are more posts than fit on the view.
     -  When I click on a post, it redirects me to that post's show page.

- Post show page:
    -  I can see the post's title.
    - I can see who wrote the post.
   -  I can see how many comments it has.
   -  I can see how many likes it has.
   -  I can see the post body.
   -  I can see the username of each commentor.
   -  I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(comment.author.name)
+    end
+    it 'I can see the comment each commentor left.' do
+      comment = Comment.create(author_id: user.id, post_id: subject.id, text: 'Nice post')
+      visit user_post_path(user.id, subject.id)
+      page.has_content?(comment.text)
+    end
+  end
+end
```
Comment (CollinsTatang) [inline spec/features/posts_spec.rb]: Hi @megha-n-bodke 
You have done a great job implementing the different User post tests

- [ ] Please create a separate file for post index page and post show page

- Has featured tests for User post index page:

- Has featured tests for User post show page:

File: spec/features/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      page.has_content?(post.title)
+    end
+    it "When I click to see all posts, it redirects me to the user's post's index page." do
+      visit user_path(subject.id)
+      click_on 'Show all Posts'
+      visit user_posts_path(subject.id)
+      page.has_content?('John')
+    end
+  end
+end
```
Comment (CollinsTatang) [inline spec/features/user_spec.rb]: Hi @megha-n-bodke 
You have done a great job implementing the different User tests

- [ ] Please create a separate file for User index page and User show page

- Has featured tests for User index page:

- Has featured tests for User show page:

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @megha-n-bodke , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**